api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -83,6 +83,18 @@ default IncrementalChangelogScan newIncrementalChangelogScan() {
     throw new UnsupportedOperationException("Incremental changelog scan is not supported");
   }
 
+  /**
+   * Create a new {@link PartitionStatisticsScan} for this table.
+   *
+   * <p>Once a partition statistics scan is created, it can be refined to project columns and filter
+   * data.
+   *
+   * @return a partition statistics scan for this table
+   */
+  default PartitionStatisticsScan newPartitionStatisticsScan() {
```

Comment:
Do we really need this method here? 

Why can't we have this method in `PartitionStatsHandler`? `PartitionStatsHandler.scanPartitionStatistics` that accepts filter, project and snapshot id and returns the `PartitionStatisticsScan`

Because, the write functionality still depends on `PartitionStatsHandler` class and if we keep a static method for scan there, it will be less disruptive change. Just for scan API, moving the classes to different module seems like an overkill to me. 

If I understand correctly, the previous implementation was not having an option to filter the output. We can just add that functionality without a major refactoring?